rotkehlchen/chain/ethereum/utils.py

```diff
@@ -208,11 +212,25 @@ def try_download_ens_avatar(
     if avatar_url == '':
         return  # Avatar is not set
 
-    try:
-        avatar = requests.get(avatar_url, timeout=DEFAULT_TIMEOUT_TUPLE).content
-    except RequestException as e:
-        log.error(f'Got error {str(e)} during querying ens avatar for {ens_name}.')
-        return
+    avatar = None
+    if avatar_url.startswith('eip155'):  # an NFT is set. Let's try first ENS app's own metadata
+        try:
+            avatar = requests.get(f'{ENS_METADATA_URL}/avatar/{ens_name}', timeout=DEFAULT_TIMEOUT_TUPLE).content  # noqa: E501
```

Comment:
Should you check that the response has status code 200 or at least no error and if it has headers that the content is `image/`?  Content for a page can also be the html text and I see this happening if hitting a cdn or nginx errors...

Reply:
This is not done at all wherever we query for images. Check the production code right now. But sure this is something that can be added.